apps/src/code-studio/assets/SpriteUpload.jsx

```diff
@@ -88,9 +127,30 @@ export default class SpriteUpload extends React.Component {
     this.setState({spriteAvailability: event.target.value, category: ''});
   };
 
+  handleAliasChange = event => {
+    let aliases = event.target.value;
+    aliases = aliases.split(',');
+    let processedAliases = aliases.map(alias => alias.trim());
+    this.setState({aliases: processedAliases});
+  };
+
+  generateMetadata = () => {
+    const {filename, aliases} = this.state;
+    let image = document.getElementById('sprite-image-preview');
```

Comment:
two small things here:
1. super nit: this can be a const
2. we should use a react `ref` to avoid mixing the react DOM with the browser's DOM. there's more usage info [here](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs), but the basic pieces of a ref look like this:

```javascript
class SpriteUpload extends React.Component {
  constructor(props) {
    this.imagePreviewRef = React.createRef();
  }

 generateMetadata = () => {
    const imageRef = this.imagePreviewRef.current;
    // from here, you can access properties similar to width/height
    // https://stackoverflow.com/questions/57322164/how-to-get-width-of-element-using-react-ref
  }

  render() {
    return (
      <img ref={this.imagePreviewRef} src="" />
    )
  }
}
```

Reply:
Looks like createRef isn't available until we upgrade to React 16, so I'll chase this down using the "callback refs" method.